unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/EmotesCustomization/EmotesCustomizationComponentController.cs

```diff
@@ -0,0 +1,442 @@
+                return;
+
+            emotesCustomizationDataStore.currentLoadedEmotes.Add(id);
+            EmoteCardComponentModel emoteToAdd = ParseWearableItemIntoEmoteCardModel(wearable);
+            EmoteCardComponentView newEmote = view.AddEmote(emoteToAdd);
+            newEmote.SetAsLoading(true);
+
+            if (!emotesInLoadingState.ContainsKey(id))
+                emotesInLoadingState.Add(id, newEmote);
+
+            RefreshEmoteLoadingState(id);
+
+            UpdateEmoteSlots();
+        }
+
+        internal void RemoveEmote(string id, WearableItem wearable)
+        {
+            emotesCustomizationDataStore.currentLoadedEmotes.Remove(id);
+            view.RemoveEmote(id);
+            UpdateEmoteSlots();
+        }
+
+        internal void OnAnimationAdded((string bodyshapeId, string emoteId) values, AnimationClip animationClip) { RefreshEmoteLoadingState(values.emoteId); }
+
+        internal void RefreshEmoteLoadingState(string emoteId)
+        {
+            if (emoteAnimations.ContainsKey((userProfile.avatar.bodyShape, emoteId)))
+            {
+                emotesInLoadingState.TryGetValue(emoteId, out EmoteCardComponentView emote);
+                if (emote != null)
+                {
+                    emote.SetAsLoading(false);
+                    emotesInLoadingState.Remove(emoteId);
+                }
+            }
+        }
+
+        internal EmoteCardComponentModel ParseWearableItemIntoEmoteCardModel(WearableItem wearable)
+        {
+            return new EmoteCardComponentModel
+            {
+                id = wearable.id,
+                name = wearable.GetName(),
+                description = wearable.description,
+                pictureUri = wearable.ComposeThumbnailUrl(),
+                pictureSprite = wearable.thumbnailSprite,
+                isAssignedInSelectedSlot = false,
+                isSelected = false,
+                assignedSlot = -1,
+                rarity = wearable.rarity,
+                isInL2 = wearable.IsInL2(),
+                isLoading = false,
+                isCollectible = wearable.IsCollectible()
+            };
+        }
+
+        internal void ConfigureUserProfileAndCatalog(UserProfile userProfile, BaseDictionary<string, WearableItem> catalog)
+        {
+            this.userProfile = userProfile;
+            this.catalog = catalog;
+
+            this.userProfile.OnInventorySet += OnUserProfileInventorySet;
+            this.userProfile.OnUpdate += OnUserProfileUpdated;
+
+            OnUserProfileUpdated(this.userProfile);
+        }
+
+        internal void OnUserProfileUpdated(UserProfile userProfile)
+        {
+            if (string.IsNullOrEmpty(userProfile.userId))
+                return;
+
+            this.userProfile.OnUpdate -= OnUserProfileUpdated;
+            catalog.OnAdded += AddEmote;
+            catalog.OnRemoved += RemoveEmote;
+            emoteAnimations.OnAdded += OnAnimationAdded;
```

Comment:
should these be unsubscribed as well?

Reply:
It is not needed because we are already doing
```
this.userProfile.OnUpdate -= OnUserProfileUpdated;
```

the first time `OnUserProfileUpdated` is executed. This unsubscription is already avoiding to execute the whole method more than 1 times.

Apart of that, we are unsubscribing these methods in the `Dispose()`.